### PR TITLE
Add howardhinnant_date recipe

### PR DESCRIPTION
**Pull request by pidefrem**
Hello,

here is PR for https://github.com/HowardHinnant/date, the datetime library for C++, soon to be part of C++20 standard.

### CODE REVIEW

File: recipes/howardhinnant_date/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$PREFIX $SRC_DIR -DCMAKE_INSTALL_LIBDIR=lib
+make
+#cmake --build . --target testit
```
Comment (scopatz) [inline recipes/howardhinnant_date/build.sh]: Please remove dead code

File: recipes/howardhinnant_date/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/HowardHinnant/date/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
```
Comment (scopatz) [inline recipes/howardhinnant_date/meta.yaml]: Please remove the empty patches

File: recipes/howardhinnant_date/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:  # Empty host dependency section
+    - libcurl
+  run:
+    - libcurl
```
Comment (scopatz) [inline recipes/howardhinnant_date/meta.yaml]: If this doesn't rely on libcurl, you can remove these lines

Comment (pidefrem) [reply recipes/howardhinnant_date/meta.yaml]: Thanks for reviewing!
Unless `USE_SYSTEM_TZ_DB=ON` is passed to `cmake` cmd line, `libtz.so` does depend on `libcurl`.

File: recipes/howardhinnant_date/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  host:
+    - libcurl
+  run:
+    - libcurl
+
+test:
+  commands:
+    - test -d ${PREFIX}/include/date  # [unix]
+    - test -f ${PREFIX}/include/date/date.h  # [unix]
+    - test -f ${PREFIX}/lib/cmake/date/dateConfig.cmake  # [unix]
```
Comment (scopatz) [inline recipes/howardhinnant_date/meta.yaml]: Is there an lib*.so that is produced?

Comment (pidefrem) [reply recipes/howardhinnant_date/meta.yaml]: Yes, `lib64/libtz.so` on linux.

Comment (scopatz) [reply recipes/howardhinnant_date/meta.yaml]: OK great, can you please test that it exists :wink:

Comment (pidefrem) [reply recipes/howardhinnant_date/meta.yaml]: Done.

File: recipes/howardhinnant_date/conda_build_config.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+c_compiler:    # [linux]
+  - gcc        # [linux]
+cxx_compiler:  # [linux]
+  - gxx        # [linux]
```
Comment (scopatz) [inline recipes/howardhinnant_date/conda_build_config.yaml]: I don't think you need this file

**Pull request merged**